### ISSUE TITLE
feat: weekly automatic newsletter

### ISSUE THREAD

**Issue by EronAlves1996** [sibelius/ccsseraphini#1469]
# Feature Request

## Why it is needed

With the introduction of sserarticles, now we can have a selection of articles choosen by community and spread them on general public. The idea here is make weekly ´mail newsletter with most ranked articles on sserarticles by engagement on Twitter. Overall, we can include twitters on sserarank and memes, to diversify it and have a link to our Fansfy (to support @EmanuelCampos product) and related Twitter accounts.

Additionaly, what about a @sibelius selected article weekly?

## Possible implementation

This can be made by using a rank system, just like sserank. We can included the 10 most rated article on the past weekly and send them on an email (with crontab) and format them with 3 elements:

- Title
- Description
- Thumbnail

These three items can be taken automatically, on the link metadata.

**Comment by MarcusXavierr** [comment]
Do you think that we need to implement some sort of "ranking" before this feature is possible? 

How can the users let us know their email? Submitting a form on a webpage? sending a dm to sserartigos bot on twitter?

@EronAlves1996

**Comment by EronAlves1996** [comment]
@sibelius answered that more faster than I on discord.
His ideas are good, and I agree with him:

![Screenshot_20230114-111822~2](https://user-images.githubusercontent.com/2249213/212476557-5b77a488-589f-4fdf-9134-b647017ec789.png)

However, I agree with "make it simple first" approach.

Mailing requires more structure. Twitter is already set, so, making the newsletter on twitter is more simple than emailling. I don't agree with reply to twitter on sserarticles, but is potentially an spam exploit for people. It should be a form on sseraphini.cc, but can be build later.

Rank can be made with the same logic of sserarank or by combining votes on discord with likes/retweets.

**Comment by MarcusXavierr** [comment]
Ok, so we can follow these steps

- [ ] Implement a simple form of ranking on sseartigos, just using twitter views, rt, and likes, so we don't need to use a database to store any form of data
- [ ] Create a module to run this rank module and get the TOP X articles.
- [ ] Create a simple module to create a cool thread with the top posts of the week. This code can be scheduled to run every Monday at 1 PM.

And that's it. Later on, we can send DMs to the users who "subscribed" to our "newsletter".

What do you think? @EronAlves1996

**Comment by EronAlves1996** [comment]
I don't think that views can help while ranking. Only if we take in account forms of relevance for each of these interactions.
As example:
1. View - Relevance: 0,5
2. Rewteet - Relevance: 1
3. Like - Relevance: 2

Just like that.
For the other steps, lgtm
